components/net/filemanager_thread.rs

```diff
@@ -0,0 +1,145 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+use ipc_channel::ipc::{self, IpcReceiver, IpcSender};
+use net_traits::filemanager_thread::{FileManagerThreadMsg, FileManagerResult, FileManagerThreadError};
+use std::cell::RefCell;
+use std::collections::HashMap;
+use std::fs::File;
+use std::io::Read;
+use std::path::{Path, PathBuf};
+use util::thread::spawn_named;
+use uuid::Uuid;
+
+pub struct FileManager {
+    receiver: IpcReceiver<FileManagerThreadMsg>,
+    idmap: RefCell<HashMap<Uuid, String>>,
```

Comment:
Currently, the `libnfd` returns a `String` in this way: `CString::from_raw(*out_path).into_string()`. Maybe there is encoding checking stuff. I will take a look at this.